homeassistant/components/prometheus/__init__.py

```diff
@@ -418,9 +418,14 @@ def _handle_sensor(self, state):
                 break
 
         if metric is not None:
-            _metric = self._metric(
-                metric, self.prometheus_cli.Gauge, f"Sensor data measured in {unit}"
-            )
+            if unit in (None, ""):
```

Comment:
```suggestion
            if not unit:
```

Reply:
Should this also be changed here?
https://github.com/home-assistant/core/blob/f5b922761f913d2b8d75046e9e2878709550bb03/homeassistant/components/prometheus/__init__.py#L463-L465

Reply:
Thanks, I applied your suggestion in the new commit and simplified / removed the duplicate code.